sale_order_type/models/sale.py

```diff
@@ -67,41 +75,70 @@ def _compute_sale_type_id(self):
                 sale_type = record._default_type_id()
             record.type_id = sale_type
 
-    @api.onchange("type_id")
-    def onchange_type_id(self):
-        # TODO: To be changed to computed stored readonly=False if possible in v14?
-        vals = {}
+    @api.depends("type_id")
+    def _compute_warehouse_id(self):
+        res = super()._compute_warehouse_id()
         for order in self:
```

Comment:
Do this on all the computed. We don't want that this empties some fields if you empty the sales order type.

```suggestion
        for order in self.filtered("type_id"):
```